components/mpas-ocean/src/shared/mpas_ocn_diagnostics.F

```diff
@@ -667,6 +668,132 @@ subroutine ocn_relativeVorticity_circulation(relativeVorticity, circulation, nor
 
    end subroutine ocn_relativeVorticity_circulation!}}}
 
+!***********************************************************************
+!
+!  routine ocn_diagnostic_solve_wctEdge
+!
+!> \brief   Computes layer thickness at edge
+!> \author  Matt Turner
+!> \date    October 2020
+!> \details
+!>  This routine computes the diagnostic variables layerThickEdgeFlux
+!>  and layerThickEdgeMean. The Mean is a simple mean across the edge
+!>  but the Flux value can either be the mean or an upwind value,
+!>  depending on user input.
+!
+!-----------------------------------------------------------------------
+
+   subroutine ocn_diagnostic_solve_wctEdge(btrVelocity, sshCell, &
+                         bottomDepthEdge, wctEdge)!{{{
+
+      !-----------------------------------------------------------------
+      ! input variables
+      !-----------------------------------------------------------------
+
+      real (kind=RKIND), dimension(:), intent(in) :: &
+         btrVelocity,   &!< [in] barotropic velocity
+         sshCell,       &!< ssh at cells, used for centered water column
+                         !< thickness
+         bottomDepthEdge !< bottom depth centered at edges, used for
+                         !< centered water column thickness
+      ! bottomDepth is already present
+
+      !-----------------------------------------------------------------
+      ! output variables
+      !-----------------------------------------------------------------
+
+      ! Outputs are the shared variables from diagnostic_variables:
+      real (kind=RKIND), dimension(:), intent(inout) :: &
+         wctEdge !< [out] wct on edge
+
+      !-----------------------------------------------------------------
+      ! local variables
+      !-----------------------------------------------------------------
+
+      integer :: &
+         iCell, iEdge, &! edge and cell loop indices
+         cell1, cell2  ! neighbor cell indices across an edge
+
+      real (kind=RKIND), dimension(:), allocatable :: wctCell, sshEdge
+
+      ! End preamble
+      !-----------------------------------------------------------------
+      ! Begin code
+
+      wctEdge(:) = -1.0e34_RKIND
+      ! Compute edge flux based on option set on init
+      select case (thickEdgeFluxChoice)
+
+      case (thickEdgeFluxCenter)
+         ! Use centered (mean) thickness as flux value
+
+         ! Compute mean layer thickness at edge
+#ifdef MPAS_OPENACC
+         !$acc parallel loop collapse(2) &
```

Comment:
I think various things may be missing here, though I can't claim particular OpenACC expertise:
```suggestion
         !$acc parallel loop &
         !$acc    present(bottomDepthEdge, sshCell, sshEdge, cellsOnEdge, wctEdge) &
         !$acc    private(cell1, cell2)
```

Reply:
Worth getting input from @philipwjones or someone else that has better OpenACC experience than I do.